### PR TITLE
Remove commons-httpclient 3.1 library dependency. 

### PR DESCRIPTION
**Pull request by abhioncbr**
As per the [issue](https://github.com/apache/pinot/issues/11118)

### CODE REVIEW

**Comment by abhioncbr** [comment]
@Jackie-Jiang, please let me know if I need to cut this PR into small ones. This PR also covers the changes of the [PR](https://github.com/apache/pinot/pull/11246)

File: pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java [hunk 1]
```diff
@@ -240,10 +230,6 @@ public static void installDefaultSSLSocketFactory(String keyStoreType, String ke
       // HttpsURLConnection
       HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
 
-      // Apache HTTP client 3.x
-      Protocol.registerProtocol("https",
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java:244]: Is it safe to remove this?

**Review by xiangfu0 — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java (hunk 1 above)
Comment (abhioncbr) [reply pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java:244]: Yes, according to the conversation [here](https://github.com/apache/pinot/pull/11246). We are removing `commons-httpclient 3.1` and this set custom `sslcontext` in this way.

**Review by Jackie-Jiang — APPROVED** [review]
Thanks for taking time cleaning this up. LGTM with minor comments

File: pinot-common/src/main/java/org/apache/pinot/common/http/MultiHttpRequestResponse.java [hunk 2]
```diff
@@ -18,38 +18,26 @@
  */
 package org.apache.pinot.common.http;
 
+import java.io.IOException;
 import java.net.URI;
 import org.apache.http.client.methods.CloseableHttpResponse;
-
-
 public class MultiHttpRequestResponse implements AutoCloseable {
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/http/MultiHttpRequestResponse.java:24]: (minor) Can we add some empty lines between methods? Auto-reformat should do the job

**Pull request merged**